platform/linux-generic/odp_packet.c

```diff
@@ -1693,7 +1693,10 @@ void odp_packet_print_data(odp_packet_t pkt, uint32_t offset,
 
 int odp_packet_is_valid(odp_packet_t pkt)
 {
-	if (odp_buffer_is_valid(packet_to_buffer(pkt)) == 0)
+	if (pkt == ODP_PACKET_INVALID)
+		return 0;
+
```

Comment:
Packet subtype validity could be checked here.

Reply:
done in v3